source/extensions/common/aws/credentials_provider_impl.cc

```diff
@@ -195,8 +195,22 @@ std::string getEnvironmentVariableOrDefault(const std::string& variable_name,
 void CredentialsFileCredentialsProvider::refresh() {
   ENVOY_LOG(debug, "Getting AWS credentials from the credentials file");
 
-  const auto credentials_file = getEnvironmentVariableOrDefault(
-      AWS_SHARED_CREDENTIALS_FILE, DEFAULT_AWS_SHARED_CREDENTIALS_FILE);
+  // Default path plus current home directory. Will fall back to / if HOME environment variable does
+  // not exist
+  std::string credential_path;
+  const char* home = getenv("HOME");
+
+  if (home) {
+    credential_path = std::string(home).append(DEFAULT_AWS_SHARED_CREDENTIALS_FILE);
+  } else {
+    credential_path = std::string(DEFAULT_AWS_SHARED_CREDENTIALS_FILE);
+  }
+
```

Comment:
Is ot possible to utilize the func getEnvironmentVariableOrDefault like below?
```
const auto home = getEnvironmentVariableOrDefault("HOME", "");
const auto default_credentials_file_path = absl::StrCat(home, DEFAULT_AWS_SHARED_CREDENTIALS_FILE);
```

Reply:
yah this is a nicer way. implemented.